demisto_sdk/commands/common/docker_helper.py

```diff
@@ -235,13 +239,21 @@ def create_container(
         """
         Creates a container and pushing requested files to the container.
         """
-        container: docker.models.containers.Container = (
-            init_global_docker_client().containers.create(
-                image=image, command=command, environment=environment, **kwargs
+        docker_client = init_global_docker_client()
+
+        try:
+            container: docker.models.containers.Container = (
+                docker_client.containers.create(
+                    image=image, command=command, environment=environment, **kwargs
+                )
             )
-        )
+        except DockerException as e:
+            container.remove()
```

Comment:
i dont think it makes sense, the `container` variable would not be defined if an exception would be raised.
I suggest to get the container from a custom name that we used / from the image it was created with.

any-way during the exception you will need to initialize the existing container model.